controller/resource.go

```diff
@@ -37,12 +44,67 @@ func (c *ResourceController) Read(ctx *app.ReadResourceContext) error {
 
 // Register runs the register action.
 func (c *ResourceController) Register(ctx *app.RegisterResourceContext) error {
-	// ResourceController_Register: start_implement
 
-	// Put your logic here
+	// TODO validate the PAT here
 
-	// ResourceController_Register: end_implement
-	return nil
+	var res *resource.Resource
+
+	err := application.Transactional(c.db, func(appl application.Application) error {
+
+		// Lookup or create the resource type
+		resourceType, err := appl.ResourceTypeRepository().LookupOrCreate(ctx, ctx.Payload.Type)
+		if (err != nil) {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+
+		// Lookup the parent resource if one has been specified
+		var parentResource *resource.Resource
+		if (ctx.Payload.ParentResourceID != nil) {
+			parentResource, err = appl.ResourceRepository().Load(ctx, *ctx.Payload.ParentResourceID)
+
+			if (err != nil) {
+				return err
+			}
+		}
+
+		// TODO extract the owner identity from the PAT
+		var identityID uuid.UUID
+
+		identity, err := appl.Identities().Load(ctx, identityID)
+		if (err != nil) {
+			// TODO raise an error if the identity could not be determined
+
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+
+		res = &resource.Resource{
+			ResourceID:     uuid.NewV4().String(),
+			ParentResource: parentResource, //ctx.Payload.ParentResourceID,
+			Owner:          *identity,
+			ResourceType:   *resourceType,
+			Description:    *ctx.Payload.Description,
+		}
+
+		// Create the resource
+		appl.ResourceRepository().Create(ctx, res)
+
+		return err
+	})
+
+
+	if (err != nil) {
+	  return jsonapi.JSONErrorResponse(ctx, err)
+	}
+
+	log.Debug(ctx, map[string]interface{}{
+		"resource_id":        res.ResourceID,
+    "parent_resource_id": res.ParentResource.ResourceID,
```

Comment:
`make format-go-code` ;)
Btw you can configure Gogland to do `go fmt` when saving the file